core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -113,6 +113,17 @@ class ControllerContext {
     }
   }
 
+  def replicasOnDeadBrokers(): Set[PartitionAndReplica] = {
+    partitionReplicaAssignmentUnderlying.flatMap {
+      case (topic, topicReplicaAssignment) => topicReplicaAssignment.flatMap {
+        case (partition, replicas) => replicas.collect {
+          case (replicaId) if (!isReplicaOnline(replicaId, new TopicPartition(topic, partition))) =>
+              PartitionAndReplica(new TopicPartition(topic, partition), replicaId)
+        }
+      }
+    }.toSet
```

Comment:
You can use `scala.collection.breakOut` like `container.flatMap(...)(breakOut)` when performing a `flatMap` followed by a `toXX` operations. This avoid have to create two "builders" and two collections instead of one builder and one collection.